pkg/components/internal/testutil/common.go

```diff
@@ -21,17 +21,33 @@ import (
 
 	"github.com/kinvolk/lokomotive/pkg/components"
 	"github.com/kinvolk/lokomotive/pkg/components/util"
+	"github.com/kinvolk/lokomotive/pkg/k8sutil"
 )
 
-// ConfigFromMap takes a map and a key and returns the value associated with the key. If the key
-// does not exist in that map, the function fails.
-func ConfigFromMap(t *testing.T, m map[string]string, k string) string {
-	ret, ok := m[k]
-	if !ok {
-		t.Fatalf("Config not found with filename: %q", k)
+// ConfigFromMap takes a map and a key. The function returns the YAML object associated with the
+// key. If the key does not exist in that map, the function fails.
+func ConfigFromMap(t *testing.T, m map[string]string, key k8sutil.ObjectMetadata) string {
+	for _, v := range m {
+		splittedYAML, err := k8sutil.SplitYAMLDocuments(v)
+		if err != nil {
+			t.Fatalf("Splitting YAML doc separated by '---': %v", err)
+		}
+
+		for _, val := range splittedYAML {
+			obj, err := k8sutil.YAMLToObjectMetadata(val)
```

Comment:
It seems those 2 methods from `k8sutil` could be private and we could expose 3rd one, which does both?

`SplitYAMLDocuments` seems like an implementation detail to me, we could make `YAMLToObjectMetadata` to return list of objects and handle that automagically.

Reply:
Yep that was my earlier approach but I was dissuaded from doing so. https://github.com/kinvolk/lokomotive/pull/1342#discussion_r575196880

Reply:
Yes, but this comment does not contradict this one. I said there that `SplitYAMLDocuments` is not a good name. It seems to me that `YAMLToObjectMetadata = splitYAMLDocuments + yamlToUnstructured` would be a good composition here.

Reply:
Lets add something like `YAMLToObjectMetadata` when we see that many users are doing `splitYAMLDocuments + yamlToUnstructured` on their own.

Reply:
What's the point of exposing 2 functions as an API if they are only used together right now?

Reply:
Let's not split hairs here. I think this is fine.